transport/awssqs/consumer.go

```diff
@@ -0,0 +1,242 @@
+package awssqs
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/transport"
+)
+
+// Consumer wraps an endpoint and provides and provides a handler for sqs msgs
+type Consumer struct {
+	sqsClient             SQSClient
+	e                     endpoint.Endpoint
+	dec                   DecodeRequestFunc
+	enc                   EncodeResponseFunc
+	wantRep               WantReplyFunc
+	queueURL              *string
+	dlQueueURL            *string
+	visibilityTimeout     int64
+	visibilityTimeoutFunc VisibilityTimeoutFunc
+	before                []ConsumerRequestFunc
+	after                 []ConsumerResponseFunc
+	errorEncoder          ErrorEncoder
+	finalizer             []ConsumerFinalizerFunc
+	errorHandler          transport.ErrorHandler
+}
+
+// NewConsumer constructs a new Consumer, which provides a Consume method
+// and message handlers that wrap the provided endpoint.
+func NewConsumer(
+	sqsClient SQSClient,
+	e endpoint.Endpoint,
+	dec DecodeRequestFunc,
+	enc EncodeResponseFunc,
+	wantRep WantReplyFunc,
+	queueURL *string,
+	dlQueueURL *string,
+	visibilityTimeout int64,
```

Comment:
This is a log of required parameters. Can any of these have sensible default values?